src/Azure/Orleans.ScheduledJobs.AzureStorage/AzureStorageJobShard.cs

```diff
@@ -83,31 +83,101 @@ public async Task UpdateBlobMetadata(IDictionary<string, string> metadata, Cance
 
     private async Task AppendOperation(JobOperation operation)
     {
-        var content = BinaryData.FromObjectAsJson(operation).ToString() + Environment.NewLine;
-        using var stream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(content));
+        var json = JsonSerializer.Serialize(operation, JobOperationJsonContext.Default.JobOperation);
+        var content = EncodeNetstring(json);
+        using var stream = new MemoryStream(content);
```

Comment:
All the allocs and copying here are pretty unfortunate. Also, the transcoding from string to UTF8 is unnecessary - we can serialize directly to UTF8 and save memory.